### DISCUSSION TITLE
Get texture position on the 3D model?

### DISCUSSION THREAD

**Discussion by MrSounds** [google/model-viewer#4576] [Q&A]
Is there a method to get the position of the texture on the 3D model?
Something like modelViewer.model.materials[0].pbrMetallicRoughness.baseColorTexture.texture.getPosition ?
It can be achieved by setting a hotspot at the desired point.

**Comment by elalish** [comment]
I'm not quite sure if this is what you're asking for, but you can get the UV coordinate of a picked point with https://modelviewer.dev/docs/index.html#entrydocs-annotations-methods-positionAndNormalFromPoint

**Reply by MrSounds** [reply]
Thanks for quick respons:)
I'm thinking about if there is a method for fetching the coordinate of a material. Lets say for example if I have a 3D-model of a doll and I want to fetch the coordinates of the material for the dolls shoes. This way I could make a button point there, based on the material for every model i added, instead of manually finding and placing these positions for every new model.

**Reply by elalish** [reply]
No, I'm not even sure how one would define such a thing, since a material can span such an arbitrary shape in space.
